tests/test_backend_features.py

```
@@ -0,0 +1,166 @@
+    """移动到危险/根外路径被拒，源文件不动。"""
+    src = _p(live_server.root, 'cover.jpg')
+    resp = requests.post(live_server.url + '/api/move',
+                         data={'src': src, 'dst_dir': _p(live_server.root, '..')})
+    assert resp.status_code == 403
+    assert os.path.exists(src)
+
+
+# ===== 安全护栏（rename/move/delete 穿越到根外被拒） =====
+
+def test_guard_traversal_outside_root(live_server, tmp_path):
+    """在 root 外造一个文件，用 .. 穿越去 rename/move/delete，都应被拒，外部文件不动。"""
```

Ruff (0.15.20)

[warning] Docstring contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF002)